java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -356,57 +368,70 @@ private API getXPAPI(ExpressionTree et, VisitorState state) {
     return API.UNKNOWN;
   }
 
+  private Matcher<MethodInvocationTree> methodRecordMatcher(
+      MethodRecord methodRecord, VisitorState state) {
+
+    return allOf(
```

Comment:
Maybe worth a few documenting comments for each of these clauses?
```
return allOf(
       // Method name must match methodRecord.getMethodName()
       (methodRecord.isStatic() ? staticMethod().anyClass() : instanceMethod().anyClass())
            .named(methodRecord.getMethodName()),
        // Method must have flag at argument index specified by record (if any)
        methodRecord
            .getArgumentIdx()
            .map(idx -> argument(idx, isArgumentMatchesFlagName(state)))
            .orElseGet(Matchers::anything),
        // Method's receiver must match record's receiver type (if any)
        methodRecord
            .getReceiverType()
            .map(
                type ->
                    receiverOfInvocation((receiver, st) -> isSameType(type).matches(receiver, st)))
            .orElseGet(Matchers::anything),
        // Method's return must match record's return type (if any)
        methodRecord.getReturnType().map(Matchers::isSameType).orElseGet(Matchers::anything));
```

I'd also consider flipping the check for receiver type and that of the argument, unless there is a performance reason for the order? (seems more natural to check method name and receiver type before the arguments)

Reply:
done!